Tests/CommonTests/AssemblyResolverTests.cs

```diff
@@ -20,145 +20,244 @@ public class AssemblyResolverTests
-            
+            Assembly testAssembly = CompileSimpleAssembly("VersionAsm1.dll", testFolder, new TestLogger(), "2.1.0.4");
+
+            // Act
+            Assembly resolvedAssembly = AssertAssemblyLoadSucceedsOnlyWithResolver("VersionAsm1, Version = 2.1.0.4, Culture = neutral, PublicKeyToken = null", testFolder);
+
+            // Assert
+            AssertExpectedAssemblyLoaded(testAssembly, resolvedAssembly);
+
+            AssertAssemblyLoadFails("VersionAsm1, Version = 1.0.0.4, Culture = neutral, PublicKeyToken = null");
+
+        }
+
+        #endregion
+
+        #region Private methods
+
+        private Assembly CompileSimpleAssembly(string assemblyFileName, string asmFolder, ILogger logger, string version = "1.0.0.0")
+        {
+            Directory.CreateDirectory(asmFolder);
+            string fullAssemblyFilePath = Path.Combine(asmFolder, assemblyFileName);
+
+            return CompileAssembly(@"public class SimpleProgram {
+              public static void Main(string[] args) {
+                System.Console.WriteLine(""Hello World"");
+              }
+            }", fullAssemblyFilePath, version, logger);
+        }
+
+        /// <summary>
+        /// Compiles the supplied code into a new assembly
+        /// </summary>
+        private static Assembly CompileAssembly(string code, string outputFilePath, string asmVersion, ILogger logger)
+        {
+            string versionedCode = string.Format(System.Globalization.CultureInfo.InvariantCulture,
+                @"[assembly:System.Reflection.AssemblyVersionAttribute(""{0}"")]
+{1}", asmVersion, code);
+
+            CompilerResults result = null;
+            using (CSharpCodeProvider provider = new CSharpCodeProvider())
+            {
+                CompilerParameters options = new CompilerParameters();
+                options.OutputAssembly = outputFilePath;
+                options.GenerateExecutable = true;
+                options.GenerateInMemory = false;
+
+                result = provider.CompileAssemblyFromSource(options, versionedCode);
+
+                if (result.Errors.Count > 0)
+                {
+                    foreach (string item in result.Output)
+                    {
+                        logger.LogInfo(item);
+                    }
+                    Assert.Fail("Test setup error: failed to create dynamic assembly. See the test output for compiler output");
+                }
+            }
+
+            return result.CompiledAssembly;
+        }
+
+        #endregion
+
+        #region Checks
+
+        private static void AssertAssemblyLoadFails(string asmRef)
+        {
+            AssertException.Expect<FileNotFoundException>(() => Assembly.Load(asmRef));
+        }
+
+        private Assembly AssertAssemblyLoadSucceedsOnlyWithResolver(string asmRef, string searchPath)
+        {
+            // Check the assembly load fails without the assembly resolver
+            AssertAssemblyLoadFails(asmRef);
```

Comment:
Checks that resolution will fail without the resolver - means we know that if resolution succeeds it is due to the resolver.